src/goDocumentation.ts

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import path = require('path');
+import { CancellationToken, TextDocumentContentProvider, Uri } from 'vscode';
+import { getBinPath, getToolsEnvVars, byteOffsetAt, getFileArchive } from './util';
+
+const missingToolMsg = 'Missing tool: ';
+
+export function getDocumentation(): void {
+	let gogetdoc = getBinPath('gogetdoc');
+	if (!path.isAbsolute(gogetdoc)) {
+		vscode.window.showErrorMessage(missingToolMsg + 'gogetdoc');
+		return;
+	}
+
+	let ate = vscode.window.activeTextEditor;
+	if (typeof ate !== 'undefined') {
+		let document = ate.document;
+		let gogetDocsFlags = ['-u', '-json', '-modified', '-pos', document.fileName + ':#' + byteOffsetAt(document, ate.selection.active).toString()];
+		let p = cp.execFile(gogetdoc, gogetDocsFlags, {env: getToolsEnvVars()}, (err, stdout, stderr) => {
+			try {
+				let goGetDocOutput = <GoGetDocOutput>JSON.parse(stdout.toString());
+
+				// drop the stuff before /vendor/ if we have it
+				let imprt = goGetDocOutput.import.replace(/(.*\/vendor\/)?(.*)/, '$2');
+
+				_getDocumentationForImport(imprt);
+			} catch (e) {
+				_getDocumentationForImport('');
+			}
+		});
+		p.stdin.end(getFileArchive(document));
+	} else {
+		_getDocumentationForImport('');
+	}
+}
+
+function _getDocumentationForImport(imprt: string): void {
+	vscode.window.showInputBox({
+		prompt: 'Please enter a package name',
+		value: imprt,
+		placeHolder: imprt === '' ? 'no package detected' : '',
+	}).then(pkgInput => {
+		if (typeof pkgInput === 'undefined' || pkgInput === '') {
+			return;
+		}
+
+		let uri = vscode.Uri.parse('godocumentation://');
+		uri = uri.with({
+			path: pkgInput,
+		});
+
+		vscode.workspace.openTextDocument(
+			uri,
+		).then(doc => vscode.window.showTextDocument(
+			doc, vscode.window.activeTextEditor.viewColumn + 1, true),
+		);
+	});
+}
+
+export class GoDocumentationContentProvider implements TextDocumentContentProvider {
+	public provideTextDocumentContent(uri: Uri, token: CancellationToken): Thenable<string> {
+		let godoc = getBinPath('godoc');
+
+		let pkg = uri.path;
+
+		// godoc doesn't handle vendor directories correctly. So we first try
+		// the local workspace vendor directory to see if there's a vendored
+		// version of the package. If not, we'll return whatever godoc would
```

Comment:
if we didnt strip out "vendor" from the gogetdoc output, then we can skip this part, correct?

Reply:
True, but it seems that wiring that into this part of the code would be somewhat non-trivial. I suppose we could add a query parameter to the URI, something like `?vendored=1` and then look for it here. But then what do you do for packages that were entered manually instead of using the `gogetdoc` flow? The `godoc` invocation is fairly cheap, so why not just try it?

Reply:
Weird.. I just tried the Go Doc command on a symbol from a vendored package.
The import path returned by gogetdoc did not have "vendor" in it.
So the attempt to strip out "vendor" became a moot point.

Reply:
That may mean that your `gogetdoc` is not detecting the right version of the package? Could you try with this set of repos https://github.com/apiarian/vscode-go-test-docs-main ? That's what I've been  using for testing. Maybe it's too small a toy?

Reply:
So here's what I see for that repo and the latest `gogetdoc` installed:

```
~/.vscode/gopath/bin/gogetdoc -json -pos main.go:#421 | jq
{
  "name": "vendoroverride",
  "import": "github.com/apiarian/vscode-go-test-docs-main/vendor/github.com/apiarian/vscode-go-test-docs-vendoroverride",
  "pkg": "vendoroverride",
  "decl": "package vendoroverride",
  "doc": "Package vendoroverride in the gopath defines foo and bar with some docs.\nThese have been overridden in the vendor directory of the main package. We\ncould probably do the same with branches or something, but this seems a bit\nmore controllable.\n",
  "pos": ""
}
```